concrete-core/src/backends/core/implementation/engines/ggsw_ciphertext_scalar_encryption.rs

```diff
@@ -0,0 +1,184 @@
+use concrete_commons::dispersion::Variance;
+use concrete_commons::parameters::{DecompositionBaseLog, DecompositionLevelCount};
+
+use crate::backends::core::implementation::engines::CoreEngine;
+use crate::backends::core::implementation::entities::{
+    GgswCiphertext32, GgswCiphertext64, GlweSecretKey32, GlweSecretKey64, Plaintext32, Plaintext64,
+};
+use crate::backends::core::private::crypto::ggsw::StandardGgswCiphertext as ImplGgswCiphertext;
+use crate::specification::engines::{
+    GgswCiphertextScalarEncryptionEngine, GgswCiphertextScalarEncryptionError,
+};
+use crate::specification::entities::GlweSecretKeyEntity;
+
+/// # Description:
+/// Implementation of [`GgswCiphertextScalarEncryptionEngine`] for [`CoreEngine`] that operates on
+/// 32 bits integers.
+impl GgswCiphertextScalarEncryptionEngine<GlweSecretKey32, Plaintext32, GgswCiphertext32>
+    for CoreEngine
+{
+    /// # Example:
+    /// ```
+    /// use concrete_commons::dispersion::Variance;
+    /// use concrete_commons::parameters::{
+    ///     DecompositionBaseLog, DecompositionLevelCount, GlweDimension, PolynomialSize,
+    /// };
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(4);
+    /// let level = DecompositionLevelCount(1);
+    /// let base_log = DecompositionBaseLog(4);
+    /// // Here a hard-set encoding is applied (shift by 20 bits)
+    /// let input = 3_u32 << 20;
+    /// let noise = Variance(2_f64.powf(-25.));
+    ///
```

Comment:
Missing comments

Reply:
There are the same amount of comments in similar functions for GLWE/LWE encryption. I suggest we keep consistency accross the doc.